calculator/calculator.py

```diff
@@ -1,15 +1,159 @@
 import re
+import math
+from statistics import mean
+from typing import Union
 
 
-class Calculator:
+__TOKEN_SPECIFICATION = {
+    'ADDITION': r'\+',
+    'SUBTRACTION': r'-',
+    'MULTIPLICATION': r'\*',
+    'DIVISION': r'/',
+    'POWER': r'\^',
+    'NEGATE': r'~',
+    'MODULE': r'%',
+    'FACTORIAL': r'!',
+    'AVERAGE': r'@',
+    'MAX': r'\$',
+    'MIN': r'&',
+    'LEFT_PARENTHESIS': r'\(',
+    'RIGHT_PARENTHESIS': r'\)',
+    'NUMBER': '\w+',
+    'NUMBER_FLOAT': '\w+\.\w+',
+}
 
-    @classmethod
-    def calculate(cls, expression=str):
-        if type(expression) is not str:
-            raise TypeError("Expression input should be of str type.")
+__NUMBER_REGEX = f'{__TOKEN_SPECIFICATION["SUBTRACTION"]}{__TOKEN_SPECIFICATION["NUMBER_FLOAT"]}|' \
+                 f'{__TOKEN_SPECIFICATION["SUBTRACTION"]}{__TOKEN_SPECIFICATION["NUMBER"]}|' \
+                 f'{__TOKEN_SPECIFICATION["NUMBER_FLOAT"]}|' \
+                 f'{__TOKEN_SPECIFICATION["NUMBER"]}'
```

Comment:
Try to use the '?' in regex, it let you make stuff optional

Reply:
I specifically want to match if any of the 4 options is present. Not sure i understand how do i use optional (?) here.

Reply:
essentialy you try to match a number or a negative number.
You can construct a number to be with optional dot and numbers after it.
And optional minus sign